src/parser/mod.rs

```diff
@@ -4139,6 +4160,167 @@ impl<'a> Parser<'a> {
+        match self.parse_one_of_keywords(&[
+            Keyword::INSERT,
+            Keyword::UPDATE,
+            Keyword::DELETE,
+            Keyword::TRUNCATE,
+        ]) {
+            Some(Keyword::INSERT) => Ok(TriggerEvent::Insert),
+            Some(Keyword::UPDATE) => {
+                if self.parse_keyword(Keyword::OF) {
+                    let cols =
+                        self.parse_comma_separated(|ident| Parser::parse_identifier(ident, false))?;
+                    Ok(TriggerEvent::Update(cols))
+                } else {
+                    Ok(TriggerEvent::Update(vec![]))
+                }
+            }
+            Some(Keyword::DELETE) => Ok(TriggerEvent::Delete),
+            Some(Keyword::TRUNCATE) => Ok(TriggerEvent::Truncate),
+            _ => self.expected(
+                "an `INSERT`, `UPDATE`, `DELETE` OR `TRUNCATE`",
+                self.peek_token(),
+            ),
+        }
+    }
+
+    pub fn parse_trigger_referencing(&mut self) -> Result<Option<TriggerReferencing>, ParserError> {
+        let refer_type = match self.parse_one_of_keywords(&[Keyword::OLD, Keyword::NEW]) {
+            Some(Keyword::OLD) if self.parse_keyword(Keyword::TABLE) => {
+                TriggerReferencingType::OldTable
+            }
+            Some(Keyword::NEW) if self.parse_keyword(Keyword::TABLE) => {
+                TriggerReferencingType::NewTable
+            }
+            _ => {
+                return Ok(None);
+            }
+        };
+
+        let is_as = self.parse_keyword(Keyword::AS);
+        let transition_relation_name = self.parse_object_name(false)?;
+        Ok(Some(TriggerReferencing {
+            refer_type,
+            is_as,
+            transition_relation_name,
+        }))
+    }
+
+    pub fn parse_trigger_exec_body(&mut self) -> Result<TriggerExecBody, ParserError> {
+        let exec_type = match self.parse_one_of_keywords(&[Keyword::FUNCTION, Keyword::PROCEDURE]) {
+            Some(Keyword::FUNCTION) => ExecBodyType::Function,
+            Some(Keyword::PROCEDURE) => ExecBodyType::Proceduer,
+            _ => {
+                return self.expected("an `FUNCTION` OR `PROCEDURE`", self.peek_token());
+            }
+        };
+
+        let func_desc = self.parse_function_desc()?;
+        Ok(TriggerExecBody {
+            exec_type,
+            func_desc,
+        })
+    }
+
+    fn parse_function_desc(&mut self) -> Result<FunctionDesc, ParserError> {
+        let name = self.parse_object_name(false)?;
+        let args = if self.consume_token(&Token::LParen) {
+            if self.consume_token(&Token::RParen) {
+                vec![]
+            } else {
+                let args = self.parse_comma_separated(Parser::parse_function_arg)?;
+                self.expect_token(&Token::RParen)?;
+                args
+            }
+        } else {
+            vec![]
+        };
+
+        Ok(FunctionDesc { name, args })
+    }
```

Comment:
this looks similar to [parse_drop_function_desc](https://github.com/sqlparser-rs/sqlparser-rs/blob/d5a6af12fd8fae89d9e861dc143dafd5378a67ab/src/parser/mod.rs#L4846C8-L4846C32) can we reuse the logic for both if so?
I wonder if it would make sense to reuse the `DropFunctionDesc` as well (we might then be able to justify renaming the type to a generic `FunctionDesc`)

Reply:
Yeah the two things are basically twins, I will proceed to keep only the generic `FunctionDesc`.

Reply:
Done - uniformed the two and updated testing accordingly.

Reply:
Maybe it should also be uniformed for `CreateFunction`, but I believe such refactoring would be outside the scope of this PR.